packages/xo-web/src/xo-app/backup-ng/new/index.js

```diff
@@ -237,87 +213,59 @@ export default [
           }
         }
 
-        const newSettings = {}
-        if (!isEmpty(state.newSchedules)) {
-          await Promise.all(
-            map(state.newSchedules, async schedule => {
-              const scheduleId = (await createSchedule(props.job.id, {
-                cron: schedule.cron,
-                timezone: schedule.timezone,
-              })).id
-              newSettings[scheduleId] = {
-                exportRetention: schedule.exportRetention,
-                copyRetention: schedule.copyRetention,
-                snapshotRetention: schedule.snapshotRetention,
-              }
-            })
-          )
-        }
-
         await Promise.all(
           map(props.schedules, oldSchedule => {
-            const scheduleId = oldSchedule.id
-            const newSchedule = find(state.schedules, { id: scheduleId })
-
-            if (
-              newSchedule !== undefined &&
-              newSchedule.cron === oldSchedule.cron &&
-              newSchedule.timezone === oldSchedule.timezone
-            ) {
-              return
-            }
+            const id = oldSchedule.id
+            const newSchedule = state.schedules[id]
 
             if (newSchedule === undefined) {
-              return deleteSchedule(scheduleId)
+              return deleteSchedule(id)
             }
 
-            return editSchedule({
-              id: scheduleId,
-              jobId: props.job.id,
-              cron: newSchedule.cron,
-              timezone: newSchedule.timezone,
-            })
+            if (
+              newSchedule.cron !== oldSchedule.cron ||
+              newSchedule.timezone !== oldSchedule.timezone
```

Comment:
`newSchedule` should simply contains the values that has been edited by the user.